frontend/appflowy_flutter/lib/shared/icon_emoji_picker/recent_icons.dart

```diff
@@ -0,0 +1,97 @@
+import 'dart:convert';
+
+import 'package:appflowy/shared/icon_emoji_picker/icon.dart';
+import 'package:appflowy_backend/log.dart';
+import 'package:flutter/foundation.dart';
+
+import '../../core/config/kv.dart';
+import '../../core/config/kv_keys.dart';
+import '../../startup/startup.dart';
+import 'flowy_icon_emoji_picker.dart';
+
+class RecentIcons {
+  static final Map<String, List<String>> _dataMap = {};
+  static bool _loaded = false;
+  static const maxLength = 20;
+
+  ///prevent the Recent Icon feature from affecting the unit tests of the Icon Selector.
+  @visibleForTesting
+  static bool enable = true;
+
+  static Future putEmoji(String id) async {
+    await _put(FlowyIconType.emoji, id);
+  }
+
+  static Future putIcon(Icon icon) async {
+    await _put(
+      FlowyIconType.icon,
+      jsonEncode(
+        Icon(name: icon.name, keywords: icon.keywords, content: icon.content)
+            .toJson(),
+      ),
+    );
+  }
+
+  static Future<List<String>> getEmojiIds() async {
+    await _load();
+    return _dataMap[FlowyIconType.emoji.name] ?? [];
+  }
+
+  static Future<List<Icon>> getIcons() async {
+    await _load();
+    final iconList = _dataMap[FlowyIconType.icon.name] ?? [];
+    try {
+      return iconList
+          .map((e) => Icon.fromJson(jsonDecode(e) as Map<String, dynamic>))
+          .toList();
```

Comment:
Do we need to deserialize from JSON every time we call getIcons? We could consider doing this once instead.

Reply:
There is a variable `_loaded` here, which is used to control the operation of deserialization only once. Moreover, the amount of data related to recent emojis is quite small, so there is no need to worry about its performance overhead for now.